includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -0,0 +1,162 @@
+		return array_slice( $items, 0, 1, true ) + $custom_endpoints + array_slice( $items, 1, null, true );
+	}
+
+	/**
+	 * Redirect to "Account details" if accessing "My Account" directly.
+	 */
+	public static function redirect_to_account_details() {
+		if ( Donations::is_platform_stripe() && function_exists( 'wc_get_page_permalink' ) ) {
+			global $wp;
+			$current_url               = home_url( $wp->request );
+			$my_account_page_permalink = wc_get_page_permalink( 'myaccount' );
+			if ( trailingslashit( $current_url ) === trailingslashit( $my_account_page_permalink ) ) {
+				wp_safe_redirect( '/my-account/edit-account/' );
+				exit;
+			}
+		}
+	}
+
+	/**
+	 * Get current user's Stripe customer ID.
+	 */
+	private static function get_current_user_stripe_id() {
+		if ( self::$stripe_customer_id ) {
+			return self::$stripe_customer_id;
+		}
+		$user_id               = get_current_user_id();
+		$user_meta_customer_id = get_user_meta( $user_id, self::STRIPE_CUSTOMER_ID_USER_META, true );
+		if ( $user_meta_customer_id ) {
+			self::$stripe_customer_id = $user_meta_customer_id;
+			return $user_meta_customer_id;
+		}
+		$customer_orders     = wc_get_orders(
+			[
+				'customer_id' => $user_id,
+				'created_via' => WooCommerce_Connection::CREATED_VIA_NAME,
+				'limit'       => -1,
+			]
+		);
+		$stripe_customer_ids = [];
+		foreach ( $customer_orders as $order ) {
+			$stripe_customer_id = $order->get_meta( '_stripe_customer_id' );
+			if ( $stripe_customer_id ) {
+				$stripe_customer_ids[] = $stripe_customer_id;
+			}
+		}
+		array_unique( $stripe_customer_ids );
+		if ( empty( $stripe_customer_ids ) ) {
+			self::$stripe_customer_id = false;
+		} else {
+			self::$stripe_customer_id = $stripe_customer_ids[0];
+			update_user_meta( $user_id, self::STRIPE_CUSTOMER_ID_USER_META, self::$stripe_customer_id );
+		}
+		return self::$stripe_customer_id;
+	}
+
+	/**
+	 * Render custom "My Account" page.
+	 */
+	public static function render_billing_template() {
+		$stripe_customer_id        = self::get_current_user_stripe_id();
+		$stripe_billing_portal_url = Stripe_Connection::get_billing_portal_url( $stripe_customer_id );
+		$error_message             = false;
+		if ( is_wp_error( $stripe_billing_portal_url ) ) {
+			$error_message = $stripe_billing_portal_url->get_error_message();
+			Logger::log( 'Error getting Stripe billing portal URL: ' . wp_json_encode( $stripe_billing_portal_url ) );
+		}
+
+		include dirname( NEWSPACK_PLUGIN_FILE ) . '/includes/reader-revenue/templates/myaccount-billing.php';
+	}
+
+	/**
+	 * Add the necessary endpoints to rewrite rules.
+	 */
+	public static function add_rewrite_endpoints() {
+		if ( ! Donations::is_platform_stripe() ) {
+			return;
+		}
+		add_rewrite_endpoint( self::BILLING_ENDPOINT, EP_PAGES );
+		if ( ! get_option( '_newspack_has_set_up_custom_billing_endpoint' ) ) {
+			flush_rewrite_rules(); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.flush_rewrite_rules_flush_rewrite_rules
```

Comment:
For some reason this didn't work for me programmatically. Until I manually flushed permalinks, the Billing page was 404ing for me.

Reply:
Hm, that weird. Are you sure this line was executed? And is there any other way to flush permalinks you're aware of?

Reply:
I don't know of another way to flush permalinks. At any rate, I tested on a new test site and didn't encounter this error, so let's consider it a non-issue. How I tested:

1. Create a new test site, install this PR, WooCommerce, and required extensions
2. Turn on required AMP Plus and Reader Activation flags in `wp-config.php`
3. Set up Stripe as the Reader Revenue platform
4. In a new session, made a recurring donation
5. Visited My Account > Billing, confirmed no 404